### PR TITLE
Remove `qml.vn_entanglement_entropy`

### PR DESCRIPTION
**Pull request by mudit2812**
`qml.vn_entanglement_entropy` was added a few weeks ago, but its purpose is almost meaningless, as `qml.vn_entanglement_entropy(wires0, wires1)` is equivalent to `qml.vn_entropy(wires0)` or `qml.vn_entropy(wires1)`. This PR removes `qml.vn_entanglement_entropy`, `VnEntanglementEntropyMP`, and any references to either of them.

[sc-75559]

### CODE REVIEW

File: pennylane/qinfo/transforms.py [hunk 1]
```diff
@@ -489,8 +489,7 @@ def vn_entanglement_entropy(
     :math:`\rho_B = \text{Tr}_A [\rho_{AB}]` are the reduced density matrices for each partition.
 
     .. warning::
-        The ``qml.qinfo.vn_entanglement_entropy`` transform is deprecated and will be removed in v0.40. Instead include
-        the :func:`pennylane.vn_entanglement_entropy` measurement process in the return line of your QNode.
+        The ``qml.qinfo.vn_entanglement_entropy`` transform is deprecated and will be removed in v0.40.
```
Comment (albi3ro) [inline pennylane/qinfo/transforms.py:492]:
```suggestion
        The ``qml.qinfo.vn_entanglement_entropy`` transform is deprecated and will be removed in v0.40. See the :func:`pennylane.vn_entropy` measurement instead.
```

File: pennylane/qinfo/transforms.py [hunk 2]
```diff
@@ -512,9 +511,8 @@ def vn_entanglement_entropy(
     """
 
     warnings.warn(
-        "The qml.qinfo.vn_entanglement_entropy transform is deprecated and will be removed "
-        "in v0.40. Instead include the qml.vn_entanglement_entropy measurement process in the "
-        "return line of your QNode.",
+        "The qml.qinfo.vn_entanglement_entropy transform is deprecated and will "
+        "be removed in v0.40.",
```
Comment (albi3ro) [inline pennylane/qinfo/transforms.py:515]:
```suggestion
        "be removed in v0.40. Instead include the qml.vn_entropy measurement process on one of the subsystems.",
```

**Review by JerryChen97 — APPROVED** [review]
lgtm

**Review by astralcai — APPROVED** [review]
We're keeping the one in `qml.math` for now?

**Comment by JerryChen97** [comment]
Probably the only potential concern is if there's such need from users: one needs first calc the rdm from some wires to a certain subset wires0+wires1, and then calc the EE(wires0, wires1). But in any ways there won't be much diff beyond vn_entropy; if requested we can add wrapper immediately.

**Comment by mudit2812** [comment]
> Probably the only potential concern is if there's such need from users: one needs first calc the rdm from some wires to a certain subset wires0+wires1, and then calc the EE(wires0, wires1). But in any ways there won't be much diff beyond vn_entropy; if requested we can add wrapper immediately.

@JerryChen97 , computing a reduced density matrix won't be necessary (from a user perspective). We can get the full density matrix from the state vector (using `dm = qml.math.dm_from_statevector(state)`), and then just compute the vn entropy on one of the subsystems (`qml.math.vn_entropy(dm, wires0)` or `qml.math.vn_entropy(dm, wires1)`)

**Pull request merged**